rules/CodeQuality/Rector/LogicalAnd/LogicalToBooleanRector.php

```diff
@@ -55,10 +55,31 @@ public function getNodeTypes(): array
      */
     public function refactor(Node $node): ?Node
     {
+        /** @var LogicalAnd|LogicalOr $left */
+        $left = $node->left;
```

Comment:
It seems the left/right can be any `Expr`:

https://github.com/nikic/PHP-Parser/blob/9aebf377fcdf205b2156cb78c0bd6e7b2003f106/lib/PhpParser/Node/Expr/BinaryOp.php#L9-L12

The `@var` assumptions already crashes many Rector rules causing fatal errors. This should be prevent now with `instanceof`

Reply:
I ensure that $left and $right are LogicalAnd|LogicalOr with the shouldRecursivelyCall function, however I understand your concern and will look into it later today

Reply:
If so, that should happen in code above.
The `$node` comming to `public function refactor(Node $node): ?Node` can be anything.

Reply:
I changed it so $node->left and $node->right are checked for the correct instances.

Reply:
Thank you